File: find-subarrays-with-equal-sum.rs
use std::collections::HashSet;

pub fn find_subarrays(nums: Vec<i32>) -> bool {
    let mut hash = HashSet::new();

    for i in 1..nums.len() {
        let sum = nums[i] + nums[i - 1];

        if hash.contains(&sum) {
            return true
        } else {
            hash.insert(sum);
        }
    }

    false
}

File: find-subarrays-with-equal-sum.ts
{
    function findSubarrays(nums: number[]): boolean {
        const set = new Set();

        for (let i = 1; i < nums.length; i++) {
            const sum = nums[i] + nums[i - 1]

            if(set.has(sum)) 
                return true
            else 
                set.add(sum)
        
        }

        return false

    };
    console.log(findSubarrays([4,2,4]))

}